src/main.rs

```diff
@@ -95,126 +109,207 @@ fn run(mut config: Config, options: &cli::Options) -> Result<(), Box<Error>> {
         info!("Configuration loaded from {}", config_path.display());
     };
 
-    // Create a display.
-    //
-    // The display manages a window and can draw the terminal
-    let mut display = Display::new(&config, options)?;
-
-    info!(
-        "PTY Dimensions: {:?} x {:?}",
-        display.size().lines(),
-        display.size().cols()
-    );
-
-    // Create the terminal
-    //
-    // This object contains all of the state about what's being displayed. It's
-    // wrapped in a clonable mutex since both the I/O loop and display need to
-    // access it.
-    let terminal = Term::new(&config, display.size().to_owned());
-    let terminal = Arc::new(FairMutex::new(terminal));
-
-    // Find the window ID for setting $WINDOWID
-    let window_id = display.get_window_id();
-
-    // Create the pty
-    //
-    // The pty forks a process to run the shell on the slave side of the
-    // pseudoterminal. A file descriptor for the master side is retained for
-    // reading/writing to the shell.
-    let mut pty = tty::new(&config, options, &display.size(), window_id);
-
-    // Create the pseudoterminal I/O loop
-    //
-    // pty I/O is ran on another thread as to not occupy cycles used by the
-    // renderer and input processing. Note that access to the terminal state is
-    // synchronized since the I/O loop updates the state, and the display
-    // consumes it periodically.
-    let event_loop = EventLoop::new(
-        Arc::clone(&terminal),
-        display.notifier(),
-        pty.reader(),
-        options.ref_test,
-    );
-
-    // The event loop channel allows write requests from the event processor
-    // to be sent to the loop and ultimately written to the pty.
-    let loop_tx = event_loop.channel();
-
-    // Event processor
-    //
-    // Need the Rc<RefCell<_>> here since a ref is shared in the resize callback
-    let mut processor = event::Processor::new(
-        event_loop::Notifier(event_loop.channel()),
-        display.resize_channel(),
-        options,
-        &config,
-        options.ref_test,
-        display.size().to_owned(),
-    );
-
-    // Create a config monitor when config was loaded from path
-    //
-    // The monitor watches the config file for changes and reloads it. Pending
-    // config changes are processed in the main loop.
-    let config_monitor = match (options.live_config_reload, config.live_config_reload()) {
```

Comment:
i just deleted all the config monitor stuff for simplicity during the refactor, i'll need to add it back in later